Public/Src/Engine/Scheduler/Fingerprints/PipFingerprinter.cs

```diff
@@ -248,7 +250,19 @@ protected virtual void AddWeakFingerprint(IFingerprinter fingerprinter, SealDire
         /// </summary>
         protected virtual void AddWeakFingerprint(IFingerprinter fingerprinter, Process process)
         {
-            AddFileDependency(fingerprinter, "Executable", process.Executable);
+            bool untrackedExecutable = false;
+            if (process.UntrackedPaths.Contains(process.Executable.Path) ||
+                process.UntrackedScopes.Any(scope => process.Executable.Path.IsWithin(m_pathTable, scope)))
+            {
+                // Don't record executable in weak fingerprint if executable is untracked
+                fingerprinter.Add("Executable", s_untrackedExecutable);

```

Comment:
I like Iman's proposal to unconditionally do `fingerprinter.Add("Executable", process.Executable)` here.